lib/ClangImporter/ImportDecl.cpp

```diff
@@ -3352,15 +3352,38 @@ namespace {
         }
       }
 
+      bool retunrsRetainedUnretainedDiagnosticsNeeded = false;
+      if (!isa<clang::CXXDeductionGuideDecl>(decl)) {
```

Comment:
Why do we need this condition? `CXXDeductionGuideDecl` isn't a `CXXMethodDecl`, so it seems redundant.

Reply:
I wanted to avoid CXXDeductionGuideDecl because FunctionDecl can be either CXXDeductionGuideDecl or CXXMethodDecl.
! CXXDeductionGuideDecl && CXXMethodDecl would be a top-lvel/global function for which I want to have the diagnostics.